picard/util/__init__.py

```diff
@@ -645,3 +649,15 @@ def limited_join(a_list, limit, join_string='+', middle_string='…'):
     start = a_list[:half]
     end = a_list[-half:]
     return join_string.join(start + [middle_string] + end)
+
+
+def extract_year_from_date(dt):
+    """ Extracts year from  passed in date either dict or string """
+
+    if isinstance(dt, Mapping):
```

Comment:
I would check for str here instead of mapping. Then return the parsed year. Makes the intention clear.

Reply:
In previous review above I was asked to change it to check for Mapping.